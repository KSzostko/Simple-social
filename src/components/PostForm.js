import React from 'react';
import { Field, reduxForm } from 'redux-form';
import styled from 'styled-components';
import Button from './Button';

const StyledForm = styled.form`
    display: flex;
    flex-direction: column;
    align-items: center;
`;

const StyledHeader = styled.h2`
    margin-top: 0;
    margin-bottom: 40px;
`;

const StyledWrapper = styled.div`
    width: 100%;
    margin-bottom: 20px;
    display: grid;
    grid-template-columns: auto 1fr;
`;

const StyledLabel = styled.label`
    margin-top: 10px;
    margin-right: 40px;
    font-size: 12px;
`;

const StyledInputField = styled(Field)`
    margin-left: 5px;
    padding: 5px 10px;
    border: 1px solid #BBB;
    border-radius: 4px;

    &:focus {
        border-color: #252B2D;
    }
`;

const StyledTextareaField = styled(Field)`
    resize: none;
    padding: 5px 10px;
    border: 1px solid #BBB;
    border-radius: 4px;

    &:focus {
        border-color: #252B2D;
    }
`;

const StyledButtonWrapper = styled.div`
    margin-top: 20px;
    width: 100%;
    display: flex;
    justify-content: flex-end;
`;

const StyledButton = styled(Button)`
    margin-right: 20px;
`;

let TestForm = props => {
    const { handleSubmit, closeModal } = props;

    return (
        <StyledForm onSubmit={handleSubmit}>
            <StyledHeader>Add post</StyledHeader>
            <StyledWrapper>
                <StyledLabel htmlFor="title">Title</StyledLabel>
                <StyledInputField name="title" component="input" type="text" />
            </StyledWrapper>
            <StyledWrapper>
                <StyledLabel htmlFor="body">Body</StyledLabel>
                <StyledTextareaField name="body" component="textarea" rows="6" />
            </StyledWrapper>
            <StyledButtonWrapper>
                <StyledButton onClick={closeModal}>Cancel</StyledButton>
                <Button type="submit">Save</Button>
            </StyledButtonWrapper>
        </StyledForm>
    );
}

TestForm = reduxForm({
    form: 'post',
})(TestForm);

export default TestForm;