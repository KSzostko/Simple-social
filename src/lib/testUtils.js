const fakePost = () => ({
    userId: 1,
    postId: 2,
    title: 'Post title',
    body: 'Post body',
});

const fakeComment = () => ({
    postId: 1,
    id: 1,
    name: 'Comment name',
    email: 'user@gmail.com',
    body: 'Comment body',
});

export { fakePost, fakeComment };