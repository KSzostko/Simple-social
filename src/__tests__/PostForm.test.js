import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import { fakePost } from '../lib/testUtils';
import PostForm from '../components/PostForm';

const fakeData = {
    closeModal: () => {},
    userdId: '1',
};

describe('<PostForm />', () => {
    it('renders and matches the snapshot', () => {
        const { container } = render(
            <Provider store={store}>
                <PostForm {...fakeData} />
            </Provider>
        );

        expect(container.firstChild).toMatchSnapshot();
    });
});