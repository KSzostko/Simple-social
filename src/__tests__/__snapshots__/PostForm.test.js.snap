// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostForm /> renders and matches the snapshot 1`] = `
<form
  action="#"
  class="sc-AxirZ cHkTwW"
>
  <h2
    class="sc-AxiKw gvpAQo"
  >
    Add post
  </h2>
  <div
    class="sc-AxhCb WMzQB"
  >
    <label
      class="sc-AxhUy glMlo"
      for="title"
    >
      Title
    </label>
    <input
      class="sc-fzozJi keCNm"
      id="title"
      name="title"
      value=""
    />
  </div>
  <div
    class="sc-AxhCb WMzQB"
  >
    <label
      class="sc-AxhUy glMlo"
      for="body"
    >
      Body
    </label>
    <textarea
      class="sc-AxgMl khZarX"
      id="body"
      name="body"
      rows="6"
    />
  </div>
  <div
    class="sc-AxheI ecSaBX"
  >
    <button
      class="sc-AxjAm sc-Axmtr debgrO"
      type="button"
    >
      Cancel
    </button>
    <button
      class="sc-AxjAm clzjYv"
      type="submit"
    >
      Save
    </button>
  </div>
</form>
`;
